drivers/flash/flash_stm32l5x.c

```diff
@@ -26,8 +26,10 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 #elif defined(CONFIG_SOC_SERIES_STM32L5X)
 #define STM32_SERIES_MAX_FLASH	512
 #elif defined(CONFIG_SOC_SERIES_STM32U5X)
-/* at this time stm32u5 mcus have 1MB (stm32u575) or 2MB (stm32u585) */
-#define STM32_SERIES_MAX_FLASH	2048
+/* It is used to handle the 2 banks discontinuity case, the discontinuity is not happen on STM32U5,
+ *  so define it to flash size to avoid the unexptected check.
+ */
+#define STM32_SERIES_MAX_FLASH	(CONFIG_FLASH_SIZE)
```

Comment:
STM32U575 is 1M and could be dual bank.
In both 1M and 2M cases, bank 2 start at 0x8100000. See https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32u5xx/drivers/include/stm32u5xx_hal_flash.h#L446 and ref man §7.3.1

Reply:
the ref manual says "When DUALBANK = 1 in option bytes, the bank 2 base address is 0x0808 0000 for 1-Mbyte dual-bank STM32U575/585 devices." on top of page 292, Table 53.
Do I misunderstand something?

Reply:
> Do I misunderstand something?

Nope, you're right. I was reading version 3 of the document which isn't clear on this point. In version 4, this is fully clear as you mention